src/libcrun/cgroup-resources.c

```diff
@@ -683,13 +683,14 @@ write_devices_resources (int dirfd, bool cgroup2, runtime_spec_schema_defs_linux
 
 /* use for cgroupv2 files with .min, .max, .low, or .high suffix */
 static int
-cg_itoa (char *buf, int64_t value, bool cgroup2)
+cg_itoa (char *buf, int64_t value, bool use_max)
 {
-  if (! (cgroup2 && value == -1))
-    return sprintf (buf, "%" PRIi64, value);
-
-  memcpy (buf, "max", 4);
-  return 3;
+  if (use_max && value < 0)
```

Comment:
Since this is an internal API, so to speak, I think you could remove the "use_max" argument (toggle).

Why? Handling of the `-1` value won't most likely be supported by the sysfs object itself (nobody would want to break the userspace), and since both cgroups v1 and v2 support this, then there is not a lot of need to have this parametrised, I suppose. The same logic of handling the "max" string value has been there since 2017 :smile: 

Thoughts?